### PR TITLE
Updated script to build TestFx environment as well as the document

### PR DESCRIPTION
**Pull request by vidai-msft**
## Description

This PR almost re-wrote the script to build testfx environment. It also re-ordered the preference how the connection string was built in testfx.
Previously, the environment variables were over json file which did not make much sense and users needed to restart VS every time when it was updated. In this PR, it has been reversed.

## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by isra-fel** [comment]
> Previously, the environment variables were over json file which did not make much sense and users needed to restart VS every time when it was updated. In this PR, it has been reversed.
Hi @vidai-msft , from my understanding, even if we re-order the priority, if user updates env var, they still need to restart VS, correct? I wonder what's the other reasons to change the priority.

**Comment by vidai-msft** [comment]
> > Previously, the environment variables were over json file which did not make much sense and users needed to restart VS every time when it was updated. In this PR, it has been reversed.
> 
> Hi @vidai-msft Vincent Dai FTE , from my understanding, even if we re-order the priority, if user updates env var, they still need to restart VS, correct? I wonder what's the other reasons to change the priority.

Yes, if they only have environment variables set. The reason why I changed the priority was if they have both json file and environment variables setup, then they just need to modify the json property and it will take effect immediately. With the previous sequence, it won't take any effect as environment variables has higher priority and the users have to update environment vars and restart VS.

File: documentation/testing-docs/using-azure-test-framework.md [hunk 1]
```diff
@@ -1,134 +1,153 @@
...
-		- [Environment Variables](#environment-variables)
-		- [Playback Test](#playback-test)
-		- [Record Test with Interactive login using OrgId](#record-test-with-interactive-login-using-orgid)
-		- [Record Test with ServicePrincipal](#record-test-with-serviceprincipal)
-- [Record/Playback tests](#record-or-playback-tests)
-- [Change Test Environment settings at run-time](#change-test-environment-settings-at-run-time)
-- [Troubleshooting](#troubleshooting)
-- [Supported Key Value pairs in ConnectionString](#supported-key-value-pairs-in-connectionstring)
-- [Environment Variable Reference](#supported-environment-in-test-framework)
+# Using Microsoft.Rest.ClientRuntime.Azure.TestFramework
```
Comment (isra-fel) [inline documentation/testing-docs/using-azure-test-framework.md:1]: I think a general name makes more sense here
```suggestion
# Using Azure PowerShell Test Framework
```

**Review by isra-fel — COMMENTED** [review]
LGTM

**Review by isra-fel — APPROVED** [review]

**Pull request merged**